sdk/app_cpu1/common/drv/timing_manager.c

```diff
@@ -156,8 +188,24 @@ void isr_0(void *intc_inst_ptr)
  */
 void timing_manager_clear_isr(void)
 {
-    Xil_Out32(TIMING_MANAGER_BASE_ADDR + TIMING_MANAGER_ISR_REG_OFFSET, 1);
-    Xil_Out32(TIMING_MANAGER_BASE_ADDR + TIMING_MANAGER_ISR_REG_OFFSET, 0);
+    uint32_t config_reg_addr = TIMING_MANAGER_BASE_ADDR + TIMING_MANAGER_ISR_REG_OFFSET;
+    // Set slv_reg4[0]
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) | 0x00000001));
+    // Clear slv_reg4[0]
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) & 0xFFFFFFFE));
+}
```

Comment:
Add a comment explaining what you are doing here.

I think you are trying to toggle a "clear" bit in the FPGA? And you have to manually reset it back to 0 again so that the "Clear" event does not just keep clearing?

This is not obvious

Reply:
So I think this would actually be better implemented similar to how @codecubepi did the manual triggering, so the FPGA handles it. The "clear isr" function could assert a bit in `slv_reg4` and then do something like this:

https://github.com/Severson-Group/AMDC-Firmware/blob/262fc4d8e81e9b8a17a55be051f62d302eb61f92/ip_repo/amdc_timing_manager_1.0/hdl/amdc_timing_manager_v1_0_S00_AXI.v#L591-L598

```Verilog
assign reset_sched_isr = slv_reg4[0] ^ reset_isr_ff;
always @(posedge S_AXI_ACLK, negedge S_AXI_ARESETN) begin
    if (!S_AXI_ARESETN)
        reset_isr_ff <= 0;
    else
        reset_isr_ff <= slv_reg4[0];
end
```

...and then

```C
Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) ^ 0x00000001));
```

I can add this in, I think this would still work and would be better than just manually resetting it in the C code I think.